appcues/src/main/java/com/appcues/trait/appcues/BackdropTrait.kt

```diff
@@ -11,28 +12,57 @@ import androidx.compose.foundation.layout.BoxScope
 import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
 import com.appcues.data.model.AppcuesConfigMap
 import com.appcues.data.model.getConfigColor
 import com.appcues.data.model.styling.ComponentColor
 import com.appcues.trait.AppcuesTraitAnimatedVisibility
 import com.appcues.trait.BackdropDecoratingTrait
+import com.appcues.trait.MetadataSettingTrait
+import com.appcues.ui.composables.LocalAppcuesStepMetadata
 import com.appcues.ui.composables.rememberAppcuesBackdropVisibility
 import com.appcues.ui.extensions.getColor
 
 internal class BackdropTrait(
     override val config: AppcuesConfigMap,
-) : BackdropDecoratingTrait {
+) : BackdropDecoratingTrait, MetadataSettingTrait {
 
     companion object {
 
         const val TYPE = "@appcues/backdrop"
+
+        const val METADATA_BACKGROUND_COLOR = "backgroundColor"
     }
 
-    private val backgroundColor = config.getConfigColor("backgroundColor")
+    override fun produceMetadata(): Map<String, Any?> {
+        return hashMapOf(METADATA_BACKGROUND_COLOR to config.getConfigColor("backgroundColor"))
+    }
 
     @Composable
     override fun BoxScope.BackdropDecorate(content: @Composable BoxScope.() -> Unit) {
+        val metadata = LocalAppcuesStepMetadata.current
+        val isDark = isSystemInDarkTheme()
+
+        val previousColor = remember(metadata) {
+            (metadata.previous[METADATA_BACKGROUND_COLOR] as ComponentColor?).getColor(isDark) ?: Color.Transparent
+        }
+
+        val actualColor = remember(metadata) {
+            (metadata.actual[METADATA_BACKGROUND_COLOR] as ComponentColor?).getColor(isDark) ?: Color.Transparent
+        }
+
+        // whenever metadata changes, set this to false
+        val animateToActual = remember(metadata) { mutableStateOf(false) }
```

Comment:
ok so I think I'm following, this is where the `LocalAppcuesStepMetadata` gets "observed" - via the remembers